src/Microsoft.OData.Edm/Validation/EdmErrorCode.cs

```diff
@@ -1388,67 +1388,57 @@ public enum EdmErrorCode
         /// </summary>
         AnnotationApplyToNotAllowedAnnotatable = 400,
 
-        #region JSON
+        /// <summary>
+        /// Invalid $Key value.
+        /// </summary>
+        InvalidKeyValue,
 
         /// <summary>
-        /// Invalid JSON
+        /// Invalid mulitple entity container in one schema
         /// </summary>
-        JsonInvalid,
+        SchemaCannotHaveMoreThanOneEntityContainer,
 
         /// <summary>
-        /// The specified version number is not valid.
+        /// Invalid $Collection value.
         /// </summary>
-        JsonInvalidVersionNumber,
+        InvalidCollectionValue,
+
+        #region JSON
 
         /// <summary>
-        /// Invalid JSON value
+        /// Invalid JSON
         /// </summary>
-        JsonInvalidValue,
+        InvalidJson,
 
         /// <summary>
         /// Invalid JSON value kind
         /// </summary>
-        JsonUnknownElementValueKind,
+        UnknownJsonElementValueKind,
 
         /// <summary>
         /// Invalid JSON number type
         /// </summary>
-        JsonInvalidNumberType,
+        InvalidJsonNumberType,
 
         /// <summary>
         /// Encountered a JSON element that is not supported
         /// </summary>
-        JsonUnsupportedElement,
+        UnsupportedJsonElement,
 
         /// <summary>
         /// Encountered a JSON element that was never used
         /// </summary>
-        JsonUnexpectedElement,
+        UnexpectedJsonElement,
 
         /// <summary>
         /// Excountered a JSON elemnt that was not expected.
         /// </summary>
-        JsonUnexpectedValueKind,
+        UnexpectedJsonValueKind,

```

Comment:
>UnexpectedJsonValueKind [](start = 8, length = 23)

Can this be "UnexpectedElementKind"?

Reply:
They are Different. So changed to UnexpectedValueKind.

---
In reply to: [445802004](https://github.com/OData/odata.net/pull/1829#discussion_r445802004) [](ancestors = 445802004)